apps/cli/src/commands/send/eip712.ts

```diff
@@ -0,0 +1,130 @@
+    chainId: 31337,
+    maxGasPrice: 10,
+} as const;
+
+type TypedData = {
+    domain: {
+        name: string;
+        version: string;
+        chainId: number;
+        verifyingContract: Address;
+    };
+    types: Record<string, { name: string; type: string }[]>;
+    primaryType: string;
+    message: Record<string, unknown>;
+};
+
+const fetchJSON = async <T>(url: string, options: RequestInit): Promise<T> => {
+    const response = await fetch(url, options);
+    if (!response.ok) {
+        throw new Error(`HTTP Error: ${response.status} - ${await response.text()}`);
+    }
+    return response.json();
+};
+
+const serializeBigInt = (_key: string, value: unknown): unknown =>
+    typeof value === "bigint" ? value.toString() : value;
+
+
+const createTypedDataTemplate = (chainId: number, verifyingContract: Address): TypedData => ({
+    domain: {
+        name: "Cartesi",
+        version: "0.1.0",
+        chainId,
+        verifyingContract,
+    },
+    types: {
+        EIP712Domain: [
+            { name: "name", type: "string" },
+            { name: "version", type: "string" },
+            { name: "chainId", type: "uint256" },
+            { name: "verifyingContract", type: "address" },
+        ],
+        CartesiMessage: [
+            { name: "app", type: "address" },
+            { name: "nonce", type: "uint64" },
+            { name: "max_gas_price", type: "uint128" },
+            { name: "data", type: "bytes" },
+        ],
+    },
+    primaryType: "CartesiMessage",
+    message: {},
+});
+
+/** Fetches the nonce for a given user and application */
+export const fetchNonceL2 = async (
+    user: Address,
+    application: Address,
+    sendOptions: SendOptions
+): Promise<number> => {
+    return fetchJSON<{ nonce: number }>(`${sendOptions.baseUrl}${SERVICE_CONTEXT_PATH}/nonce`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ msg_sender: user, app_contract: application }),
+    }).then((res) => res.nonce);
+};
+
+/** Submits a transaction to L2 */
+export const submitTransactionL2 = async (
+    data: Record<string, unknown>,
+    sendOptions: SendOptions
+): Promise<{ id: string }> => {
+    return fetchJSON<{ id: string }>(`${sendOptions.baseUrl}${SERVICE_CONTEXT_PATH}/submit`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(data, serializeBigInt),
+    });
+};
+
+/** Creates a Typed Data object for signing */
+export const createTypedData = async (
+    app: Address,
+    data: Hex,
+    nonce: number,
+    sendOptions: SendOptions
+): Promise<TypedData> => {
+    return {
+        ...createTypedDataTemplate(sendOptions.chainId, app),
+        message: {
+            app,
+            nonce,
+            data,
+            max_gas_price: BigInt(sendOptions.maxGasPrice),
+        },
+    };
+};
+
+/** Sends a transaction to L2 */
+export const sendToL2 = async (
```

Comment:
Given that we decided to call the command `send --type=eip712`, wouldn't it make more sense to call this method `sendEip712()` or something like that? Similar reasoning applies to other `xyzL2()` methods above